scubagoggles/rego/Drive.rego

```diff
@@ -0,0 +1,1588 @@
+#--
+
+DriveId1_3 := utils.PolicyIdWithSuffix("GWS.DRIVEDOCS.1.3")
+
+LogMessage1_3 := "SHARING_OUTSIDE_DOMAIN"
+
+Check1_3_OK if {
+    not PolicyApiInUse
+    events := utils.FilterEventsOU(LogEvents, LogMessage1_3, utils.TopLevelOU)
+    count(events) > 0
+}
+
+Check1_3_OK if {PolicyApiInUse}
+
+AcceptableValues1_3 := {"SHARING_ALLOWED_WITH_WARNING",
+                        "SHARING_NOT_ALLOWED",
+                        "INHERIT_FROM_PARENT",
+                        "SHARING_NOT_ALLOWED_BUT_MAY_RECEIVE_FILES",
+                        "TRUSTED_DOMAINS_ALLOWED_WITH_WARNING",
+                        "TRUSTED_DOMAINS_ALLOWED_WITH_WARNING_MAY_RECEIVE_FILES_FROM_ANYONE"}
+
+GetFriendlyValue1_3(Value) := "enabled" if {
+    Value in AcceptableValues1_3 == true
+} else := "disabled"
+
+NonComplianceMessage1_3(value) := sprintf("External Sharing Warning is %s",
+                                          [value])
+
+NonCompliantOUs1_3 contains {
+    "Name": OU,
+    "Value": NonComplianceMessage1_3(GetFriendlyValue1_3(LastEvent.NewValue))
+}
+if {
+    not PolicyApiInUse
+    some OU in utils.OUsWithEvents
+    Events := utils.FilterEventsOU(LogEvents, LogMessage1_3, OU)
+    count(Events) > 0
+    LastEvent := utils.GetLastEvent(Events)
+    not LastEvent.NewValue in AcceptableValues1_3
+}
+
+NonCompliantGroups1_3 contains {
+    "Name": Group,
+    "Value": NonComplianceMessage1_3(GetFriendlyValue1_3(LastEvent.NewValue))
+}
+if {
+    not PolicyApiInUse
+    some Group in utils.GroupsWithEvents
+    Events := utils.FilterEventsGroup(LogEvents, LogMessage1_3, Group)
+    count(Events) > 0
+    LastEvent := utils.GetLastEvent(Events)
+    not LastEvent.NewValue in AcceptableValues1_3
+}
+
+NonCompliantOUs1_3 contains {
+    "Name": OU,
+    "Value": NonComplianceMessage1_3(GetFriendlyEnabledValue(warnExternal))
+}
+if {
+    some OU, settings in input.policies
+    DriveEnabled(OU)
+    ExternalSharingAllowed(OU)
+    warnExternal := settings.drive_and_docs_external_sharing.warnForExternalSharing
+    warnExternal != true
+}
+
+tests contains {
+    "PolicyId": DriveId1_3,
+    "Criticality": "Shall",
+    "ReportDetails": utils.NoSuchEventDetails(DefaultSafe, utils.TopLevelOU),
+    "ActualValue": "No relevant event for the top-level OU in the current logs",
+    "RequirementMet": DefaultSafe,
+    "NoSuchEvent": true
+}
+if {
+    not PolicyApiInUse
+    DefaultSafe := true
+    not Check1_3_OK
+}
+
+tests contains {
+    "PolicyId": DriveId1_3,
+    "Criticality": "Shall",
+    "ReportDetails": utils.ReportDetails(NonCompliantOUs1_3, NonCompliantGroups1_3),
+    "ActualValue": {"NonCompliantOUs": NonCompliantOUs1_3,
+        "NonCompliantGroups": NonCompliantGroups1_3},
+    "RequirementMet": Status,
+    "NoSuchEvent": false
+}
+if {
+    Check1_3_OK
+    Conditions := {count(NonCompliantOUs1_3) == 0, count(NonCompliantGroups1_3) == 0}
+    Status := (false in Conditions) == false
+}
+
+#--
+
+#
+# Baseline GWS.DRIVEDOCS.1.4
+#--
```

Comment:
If you select "ALLOWLISTED DOMAINS" and check "Allow users or shared drives in... to share items with non-Google users in trusted domains using visitor sharing" 1.4 still passes.

If you select "ON" and check "Allow users or shared drives in...to share items with people outside...who aren't using a Google account," 1.4 fails, as it should, but the report details string is incorrect: "External Sharing is enabled but sharing items to non-google accounts is disabled" ("disabled" instead of "enabled").

Reply:
done